File: app/src/main/java/org/intelehealth/app/utilities/BaseEnum.java
package org.intelehealth.app.utilities;

import androidx.annotation.IntDef;

import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;

/**
 * Created by Prajwal Maruti Waingankar on 04-07-2022, 18:06
 * Copyright (c) 2021 . All rights reserved.
 * Email: prajwalwaingankar@gmail.com
 * Github: prajwalmw
 */

public class BaseEnum {

    public static final int NONE = -1;
    public static final int CMD_ESC = 1, CMD_TSC = 2, CMD_CPCL = 3, CMD_ZPL = 4, CMD_PIN = 5;
    public static final int CON_BLUETOOTH = 1, CON_BLUETOOTH_BLE = 2, CON_WIFI = 3, CON_USB = 4, CON_COM = 5;
    public static final int NO_DEVICE = -1, HAS_DEVICE = 1;

    @IntDef({CMD_ESC, CMD_TSC, CMD_CPCL, CMD_ZPL, CMD_PIN})
    @Retention(RetentionPolicy.SOURCE)
    public @interface CmdType {
    }

    @IntDef({CON_BLUETOOTH, CON_WIFI, CON_USB, CON_COM, NONE})
    @Retention(RetentionPolicy.SOURCE)
    public @interface ConnectType {
    }


    @IntDef({NO_DEVICE, HAS_DEVICE})
    @Retention(RetentionPolicy.SOURCE)
    public @interface ChooseDevice {
    }



}

File: app/src/main/java/org/intelehealth/app/activities/prescription/thermalprinter/BaseEnum.kt
package org.intelehealth.app.activities.prescription.thermalprinter

class BaseEnum {  companion object {
    const val NONE = -1
    const val CMD_ESC = 1
    const val CMD_TSC = 2
    const val CMD_CPCL = 3
    const val CMD_ZPL = 4
    const val CMD_PIN = 5

    const val CON_BLUETOOTH = 1
    const val CON_BLUETOOTH_BLE = 2
    const val CON_WIFI = 3
    const val CON_USB = 4
    const val CON_COM = 5

    const val NO_DEVICE = -1
    const val HAS_DEVICE = 1
}

    @Target(AnnotationTarget.TYPE, AnnotationTarget.VALUE_PARAMETER, AnnotationTarget.PROPERTY, AnnotationTarget.FIELD)
    @Retention(AnnotationRetention.SOURCE)
    annotation class CmdType

    @Target(AnnotationTarget.TYPE, AnnotationTarget.VALUE_PARAMETER, AnnotationTarget.PROPERTY, AnnotationTarget.FIELD)
    @Retention(AnnotationRetention.SOURCE)
    annotation class ConnectType

    @Target(AnnotationTarget.TYPE, AnnotationTarget.VALUE_PARAMETER, AnnotationTarget.PROPERTY, AnnotationTarget.FIELD)
    @Retention(AnnotationRetention.SOURCE)
    annotation class ChooseDevice
}